pkg/util/iptables/iptables.go

```diff
@@ -328,10 +328,11 @@ func (runner *runner) checkRuleWithoutCheck(table Table, chain Chain, args ...st
 
 	// Sadly, iptables has inconsistent quoting rules for comments.
 	// Just unquote any arg that is wrapped in quotes.
-	argsCopy := make([]string, len(args))
-	copy(argsCopy, args)
-	for i := range argsCopy {
-		unquote(&argsCopy[i])
+	var argsCopy []string
+	for i := range args {
+		tmpField := args[i]
+		unquote(&tmpField)
+		argsCopy = append(argsCopy, strings.Fields(tmpField)...)
```

Comment:
well... I'm not sure this has worked at all. (at least on the iptables version i'm using...)

Using master version of kube-proxy, (same results for the 1.0 version too)
this is the output of iptables-save -t nat right after executing kube-proxy
```
# Generated by iptables-save v1.4.7 on Wed Aug 19 09:38:41 2015
*nat
:PREROUTING ACCEPT [2953865:238247467]
:INPUT ACCEPT [3115:142808]
:OUTPUT ACCEPT [4330:299904]
:POSTROUTING ACCEPT [4330:299904]
:DOCKER - [0:0]
:KUBE-NODEPORT-CONTAINER - [0:0]
:KUBE-NODEPORT-HOST - [0:0]
:KUBE-PORTALS-CONTAINER - [0:0]
:KUBE-PORTALS-HOST - [0:0]
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CON
TAINER 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A KUBE-NODEPORT-CONTAINER -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 32749 -j REDIRECT --to-ports 39018 
-A KUBE-NODEPORT-HOST -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 32749 -j DNAT --to-destination 10.101.61.147:39018 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.135/32 -p tcp -m comment --comment "kube-system/kube-ui:" -m tcp --dport 80 -j REDIRECT --to-ports 60074 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.1/32 -p tcp -m comment --comment "default/kubernetes:" -m tcp --dport 443 -j REDIRECT --to-ports 40516 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.160/32 -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 80 -j REDIRECT --to-ports 39018 
-A KUBE-PORTALS-HOST -d 192.168.20.135/32 -p tcp -m comment --comment "kube-system/kube-ui:" -m tcp --dport 80 -j DNAT --to-destination 10.101.61.147:60074 
-A KUBE-PORTALS-HOST -d 192.168.20.1/32 -p tcp -m comment --comment "default/kubernetes:" -m tcp --dport 443 -j DNAT --to-destination 10.101.61.147:40516 
-A KUBE-PORTALS-HOST -d 192.168.20.160/32 -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 80 -j DNAT --to-destination 10.101.61.147:39018 
COMMIT
# Completed on Wed Aug 19 09:38:41 2015
```

and.. this is what happens after a few seconds.
```
# Generated by iptables-save v1.4.7 on Wed Aug 19 09:39:01 2015
*nat
:PREROUTING ACCEPT [53:4287]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
:DOCKER - [0:0]
:KUBE-NODEPORT-CONTAINER - [0:0]
:KUBE-NODEPORT-HOST - [0:0]
:KUBE-PORTALS-CONTAINER - [0:0]
:KUBE-PORTALS-HOST - [0:0]
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CONTAINER 
-A PREROUTING -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-CONTAINER 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m comment --comment "handle ClusterIPs; NOTE: this must be before the NodePort rules" -j KUBE-PORTALS-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A OUTPUT -m addrtype --dst-type LOCAL -m comment --comment "handle service NodePorts; NOTE: this must be the last rule in the chain" -j KUBE-NODEPORT-HOST 
-A KUBE-NODEPORT-CONTAINER -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 32749 -j REDIRECT --to-ports 53444 
-A KUBE-NODEPORT-HOST -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 32749 -j DNAT --to-destination 10.101.61.147:53444 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.1/32 -p tcp -m comment --comment "default/kubernetes:" -m tcp --dport 443 -j REDIRECT --to-ports 42922 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.160/32 -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 80 -j REDIRECT --to-ports 53444 
-A KUBE-PORTALS-CONTAINER -d 192.168.20.135/32 -p tcp -m comment --comment "kube-system/kube-ui:" -m tcp --dport 80 -j REDIRECT --to-ports 48642 
-A KUBE-PORTALS-HOST -d 192.168.20.1/32 -p tcp -m comment --comment "default/kubernetes:" -m tcp --dport 443 -j DNAT --to-destination 10.101.61.147:42922 
-A KUBE-PORTALS-HOST -d 192.168.20.160/32 -p tcp -m comment --comment "default/my-nginx:" -m tcp --dport 80 -j DNAT --to-destination 10.101.61.147:53444 
-A KUBE-PORTALS-HOST -d 192.168.20.135/32 -p tcp -m comment --comment "kube-system/kube-ui:" -m tcp --dport 80 -j DNAT --to-destination 10.101.61.147:48642 
COMMIT
# Completed on Wed Aug 19 09:39:01 2015
```

All the rules containing quoted comments get duped..